### PR TITLE
126 | Switching to pytest-regtest for regression tests

### LINKED ISSUE #126: Switch to pytest-regtest for regression tests
Currently `tests/test_regression.py` reads in two CSV files for comparison.

There is however a `pytest` plugin [`pytest-regtest`](https://pypi.org/project/pytest-regtest/) which formalises this procedure and handles the files against which comparison are to be made.

We should switch to using this, its not a huge amount of work and can be done quickly.

### PR DESCRIPTION
**Pull request by ns-rse**
Closes #126 

Rather than reading target CSV files from `tests/resources/data/` and comparing them to output the targets are now handled by virtue of the [pytest-regtest](https://gitlab.com/uweschmitt/pytest-regtest) plugin for `pytest`.

This provides a fixture (`regtest`) which can be used by a test and at the end a `print()` statement is made which compares the results to an existing target.  The existing target needs to be created in the first instance and this is done by first running `pytest --regtest-reset` to generate the files (and also update them in the future when you know you need to update them).

Its mostly a cosmetic change but means we have less to manage/maintain on the target CSV side.

### CODE REVIEW

**Comment by ns-rse** [comment]
I would approve this...but can't since its my PR!

**Comment by bobturneruk** [comment]
I plan to look at it.

**Review by Ankur-AVP-Patel — APPROVED** [review]

File: tests/test_regression.py [hunk 1]
```diff
@@ -19,10 +21,12 @@ def test_matching_mq_baseline(mq_test_df, theo_masses_df, mod_test, mq_baseline_
     )
     output_df = pd.read_csv(output_filepath, index_col=0)
 
-    pd.testing.assert_frame_equal(output_df, mq_baseline_df)
+    print(output_df.to_string(), file=regtest)
```
Comment (bobturneruk) [inline tests/test_regression.py:24]: Any idea why these statements fail on mismatched output? I'm sure they do, but it's hard to imagine how this has been done.

**Review by bobturneruk — APPROVED** [review]
Looks good. I trust that `pytest-regtest` does what it says.

**Pull request merged**